modules/mkdocs.py

```diff
@@ -0,0 +1,166 @@
+            site = self.get_site(name_or_alias, room=room)
+            if not site:
+                return await bot.send_text(room, event, f'Sorry, could not find settings for site or alias `{name_or_alias}`')
+            results = await self.search_index(site, args)
+            if len(results):
+                msg = ['I found these results:']
+                for (_, result) in results[:3]:
+                    msg.append(f'- [{result.title}]({site["url"]}/{result.location}')
+                # msg.append('For more results, see: ')
+                return await bot.send_text(room, event, '\n'.join(msg))
+            else:
+                return await bot.send_text(room, event, 'Unable to find a reference for the search.')
+
+    async def add_site(self, bot, room, event=None, args=[]):
+        name, url = args
+        sites = self.rooms[room.room_id]
+        if sites.get(name):
+            self.rooms[room.room_id][name]['url'] = url
+            await bot.send_text(room, event, f'Updated url for site `{name}` for searching')
+        else:
+            self.rooms[room.room_id][name] = {
+                'url': url,
+                'aliases': []
+            }
+            await bot.send_text(room, event, f'Added new site `{name}` for searching')
+        await self.populate_index(name, room=room)
+        bot.save_settings()
+
+    def get_site(self, name_or_alias, room):
+        sites = self.rooms[room.room_id]
+        if sites.get(name_or_alias):
+            return sites[name_or_alias]
+        else:
+            for site in sites:
+                if name_or_alias in site['alises']:
+                    return site
+        return None
+
+
+
+    async def search_index(self, site, keywords):
+        matches = []
+        docs = self.site_data[site.url]
+        for doc in reversed(docs):
+            weight = 0
+            for keyword in keywords:
+                if re.search(keyword, doc['text'], re.IGNORECASE):
+                    weight = weight + 1
+            if weight > 0:
+                matches.append((weight, doc))
+        if len(keywords) > 1:
+            matches.sort(key=lambda tup: tup[0])
+        return matches
+
+    async def populate_index(self, name, room):
+        url = self.rooms[room.room_id][name]['url']
+        response = requests.get(f'{url}/search/search_index.json').json()
+        self.sites[name] = {
+            'config': response['config'],
+        }
+        self.site_data[url] = []
+        for doc in response['docs']:
+            parts = doc.location.split('/')
+            if len(parts) < 2:
+                continue
+            *_, tail = parts
+            if tail.startswith("#"):
+                # MKDocs search has index for different headers. Ignore.
+                continue
+
+            self.sites[url].append(doc)
+
+
+
+    def help(self):
+        return 'Search MkDocs sites for relevent pages'
+
+    def long_help(self, bot=None, room=None, event=None):
```

Comment:
```suggestion
    def long_help(self, bot=None, room=None, event=None, args=None):
```

Needed for `!help mkdocs`, I must have added it to the upstream bot at some point. Could allow more granular help ("!help mkdocs search" would pass `args = ['search'], for example)